src/vanilla.ts

```diff
@@ -241,21 +241,21 @@ export const subscribe = (
   }
 }
 
-export type DeepResolveType<T> = T extends (...args: any[]) => any
+export type Snapshot<T> = T extends (...args: any[]) => any
```

Comment:
Please don't change the type name. We may consider renaming in the future, but not this time. Even thought it's not a documented api, changing it can be type breaking.

```suggestion
export type DeepResolveType<T> = T extends (...args: any[]) => any
```